tardis/montecarlo/base.py

```diff
@@ -97,24 +91,14 @@ def __init__(
         integrator_settings,
         v_packet_settings,
         spectrum_method,
+        packet_source,
         virtual_packet_logging,
         nthreads=1,
-        packet_source=None,
         debug_packets=False,
         logger_buffer=1,
         tracking_rpacket=False,
         use_gpu=False,
     ):
-        self.seed = seed
-        if packet_source is None:
-            if not enable_full_relativity:
```

Comment:
I don't think the current changes you've made handle the creation of relativistic black body packets. It should be chosen before the packet source is passed to the Monte Carlo transport step.

Reply:
@andrewfullard I had moved the relativistic black body packets logic to the from_config classmethod.